### PR TITLE
fix(TCOMP-2066): Job - afterGroup not called on Lifecycle::stop

### PR DESCRIPTION
**Pull request by undx**
https://jira.talendforge.org/browse/TCOMP-2066

### CODE REVIEW

File: component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/chain/AutoChunkProcessor.java [hunk 1]
```diff
@@ -55,6 +58,7 @@ public void flush(final OutputFactory outs) {
 
     @Override
     public void stop() {
+        flush(remains);
```
Comment (ozhelezniak-talend) [inline component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/chain/AutoChunkProcessor.java:61]: It looks like we are doing something wrong with this OutputFactory parameter being saved from the parameters.
Can we invoke flush before "stop()" on the client side?

File: component-runtime-manager/src/test/java/org/talend/test/FileOutput.java [hunk 2]
```diff
@@ -42,18 +47,36 @@ public class FileOutput implements Serializable {
...
-    public void length(final JsonObject data) throws IOException {
+    private List<JsonObject> buffer;
+
+    @BeforeGroup
+    public void init() {
+        buffer = new ArrayList<>();
+    }
+
+    @AfterGroup
+    public void commit(final Collection<JsonObject> records) {
```
Comment (ozhelezniak-talend) [inline component-runtime-manager/src/test/java/org/talend/test/FileOutput.java:58]: Is it possible to have in the same Output a method with ElementListener and AfterGroup method with records as an input?

Comment (undx) [reply component-runtime-manager/src/test/java/org/talend/test/FileOutput.java:58]: Yes, you can manage it manually (elementlistener will receive record by record). Here, we have the shortess one.
```
    @ElementListener
    public void bufferize(final JsonObject object) {
        buffer.add(object);
    }

    @AfterGroup
    public void commit() {
        // saves buffered records at once (bulk)
    }
``` 
or just
 ```
    @AfterGroup
    public void commit(final Collection<Record> records) {
        // saves records
    }
```

Comment (ozhelezniak-talend) [reply component-runtime-manager/src/test/java/org/talend/test/FileOutput.java:58]: yes, but in your example you use both at the same time
my question is about using it at the same time

Comment (undx) [reply component-runtime-manager/src/test/java/org/talend/test/FileOutput.java:58]: oups, missed the signature of the commit method... right

**Review by ozhelezniak-talend — APPROVED** [review]
lgtm

**Pull request merged**